server/controllers/recipeController.js

```diff
@@ -58,11 +58,38 @@ const emptyShoppingList = asyncHandler(async (req, res) => {
 	res.json({"message": "Shopping list successfully emptied"})
 })
 
+const addRecipe = asyncHandler( async (req, res) => {
+    const { recipeName, description, instructions, ingredients } = req.body
+
+    if((await getByName(recipeName, req.user.dataValues.id, Recipe)) == null){
+        if(await createNewRecipe(recipeName, description, instructions, req.user.dataValues.id, Recipe)){
+            const new_recipe = await getByName(recipeName, req.user.dataValues.id, Recipe);
```

Comment:
can this be `const new_recipe = await createNewRecipe` instead of having to get the data again after creating?

Reply:
WIP.

Reply:
Resolved.